Content.Server/Explosion/Components/ExplosiveComponent.cs

```diff
@@ -1,11 +1,21 @@
+using Content.Server.Destructible.Thresholds.Behaviors;
 using Robust.Shared.GameObjects;
 using Robust.Shared.Serialization.Manager.Attributes;
 
 namespace Content.Server.Explosion.Components
 {
     /// <summary>
-    /// Specifies an explosion range should this entity be exploded.
+    ///     Specifies an explosion range should this entity be exploded.
     /// </summary>
+    /// <remarks>
+    ///     Explosions can be caused by:
+    ///     <list type="bullet">
+    ///         <item>Reaching a damage threshold that causes a <see cref="ExplodeBehavior"/></item>
+    ///         <item>Being triggered via the <see cref="ExplodeOnTriggerComponent"/></item>
+    ///         <item>Manually by some other system via functions in <see cref="ExplosionHelper"/> (for example, see
+    ///         <see cref="WelderComponent"/>'s interaction with fuel tanks).</item>
```

Comment:
i'd like for weldercomponent to not directly invoke this, but for it to just apply heat damage and have that triggering the explosion.

Reply:
Yeah that actually makes more sense.

I've removed the direct call to SpawnExplosion and instead just added a DamageOnToolInteract component to fuel tanks, so they take damage when interacted with using a lit welder.